proxy/envoy/watcher.go

```diff
@@ -225,7 +230,9 @@ func generateCertHash(certsDir string) []byte {
 			glog.Warningf("failed to read file %q", filename)
 			continue
 		}
-		h.Write(bs)
+		if err := h.Write(bs); err != nil {
```

Comment:
I think we have two options: 1) log and ignore the error in `generateCertHash()` and the caller will try to schedule a config update. The update may fail (e.g. Envoy cannot read the certs). 2) return an error in `generateCertHash()`. The caller sees the error and stop config update and Envoy continues to use an old config/certs.

I'm leaning towards 1) to fail early.